compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -577,15 +576,24 @@ static LogicalResult setTransposeConfig(func::FuncOp entryPoint,
   int32_t tileM = 32;
   int32_t tileN = 32;
   TileSizesListType tileSizes;
-  tileSizes.push_back({tileM, tileN});
 
-  // Check alignment with tile size for each transpose.
+  auto linalgOp = dyn_cast<linalg::GenericOp>(op);
+  if (!linalgOp) {
+    return failure();
+  }
```

Comment:
nit: this seems redundant with line 548, I would suggest changing this function signature to pass directly a linalg::GeneircOp, this way it would avoid confusion.